File: PFinalQytetetJava/QytetetJava/src/modeloqytetet/Sorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author antonio-w10
 */
public class Sorpresa {
    
    private String texto;
    private TipoSorpresa tipo;
    private int valor;
    
    public Sorpresa(String texto, int valor, TipoSorpresa tipo){
        this.texto=texto;
        this.valor=valor;
        this.tipo=tipo;
        
    }
    
    String getTexto(){
        return texto;
    }
    
    TipoSorpresa getTipo(){
        return tipo;
    }
    int getValor(){
        return valor;
    }
    
    @Override
    public String toString() {
        return "Sorpresa{" + "texto=" + texto + ", valor=" + 
        Integer.toString(valor) + ", tipo=" + tipo + "}";
    } 
    
    
    

    
    
    
    
}

File: PFinalQytetetRuby/QytetetRuby/lib/sorpresa.rb
#encoding: UTF-8

require_relative "tipo_sorpresa"

module ModeloQytetet
  class Sorpresa
    attr_reader :texto, :tipo, :valor
    def initialize(texto, valor, tipo)
      @texto = texto
      @valor = valor
      @tipo = tipo
    
    end
    
    def to_s
      "Texto:#{@texto} Valor: #{@valor} Tipo: #{@tipo}"
    end 
  end
end

